nodes/src/nodes/visual_similarity_filter/IInstance.py

```
@@ -0,0 +1,180 @@
+    # Per-frame similarity check + ring buffer forwarding
+    # ------------------------------------------------------------------
+
+    def _on_frame_complete(self, image_bytes: bytes, mime: str):
+        idx = self._frame_idx
+        self._frame_idx += 1
+        self._total += 1
+        timestamp = idx / self._fps if self._fps > 0 else float(idx)
+
+        try:
+            is_match, similarity = self.IGlobal.embedder.is_match(image_bytes)
+        except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)